x/concentrated-liquidity/math/math.go

```diff
@@ -90,9 +101,18 @@ func CalcAmount1Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 	// additionally, without rounding, there exists cases where the swapState.amountSpecifiedRemaining.GT(sdk.ZeroDec()) for loop within
 	// the CalcOut/In functions never actually reach zero due to dust that would have never gotten counted towards the amount (numbers after the 10^6 place)
 	if roundUp {
-		return liq.Mul(diff).Ceil()
+		// Note that we do MulRoundUp so that the end result is larger as this is
+		// the case where we want to round up to favor the pool.
+		// Examples include:
+		// - calculating amountIn during swap
+		// - adding liquidity (request user to provide more tokens in in favor of the pool)
+		return liq.MulRoundUp(diff).Ceil()
 	}
-	return liq.Mul(diff)
+	// This is truncated at precision end to round in favor of the pool when:
+	// - calculating amount out during swap
+	// - withdrawing liquidity
+	// The denominator is rounded up to get a higher final amount.
+	return liq.MulTruncate(diff)
```

Comment:
Same point as above, but since this is the final output it might make sense to truncate. Still not 100% sure on what the source of the rounding error here would be, but I might have a misunderstanding of how `Mul` works.

Reply:
I'm pretty sure this matters at the precision end for the reasons in the earlier thread.

Let me resolve this one but keep the one above open